lib/stdext.ml

```diff
@@ -10,3 +10,24 @@ module Sbytes = struct
     try R.ok (Bytes.blit_string src srcoff dst dstoff len)
     with Invalid_argument e -> R.error_msg e
 end
+
+module Fpath = struct
+  let is_backup_file str =
+    let len = String.length str in
+    len > 0
+    &&
+    let first = str.[0] in
+    let last = str.[len - 1] in
+    Char.equal last '~' || (Char.equal first '#' && Char.equal last '#')
+
+  let find_files ~name_wo_ext path =
+    let open Fpath in
+    OS.Dir.contents path >>| fun files ->
```

Comment:
What would you think about making this function pure by passing it the file list so we can test the logic?

Reply:
great idea, done